services/ymax-planner/src/engine.ts

```diff
@@ -492,22 +493,41 @@ export const processPendingTxEvents = async (
     let data;
     try {
       // Extract txId from path (e.g., "published.ymax0.pendingTxs.tx1")
-      const txId = stripPrefix(`${pendingTxPathPrefix}.`, path);
+      const txId = stripPrefix(`${pendingTxPathPrefix}.`, path) as TxId;
       log('Processing pendingTx event', path);
 
       const streamCell = parseStreamCell(cellJson, path);
       const value = parseStreamCellValue(streamCell, -1, path);
       data = marshaller.fromCapData(value);
-      if (
-        data?.status !== TxStatus.PENDING ||
-        data.type === TxType.CCTP_TO_AGORIC
-      )
+
+      // If transaction is no longer PENDING, abort any ongoing watchers
+      if (data?.status !== TxStatus.PENDING) {
+        const abortController = pendingTxAbortControllers.get(txId);
+        if (abortController) {
+          log(
+            `Aborting watcher for ${txId} - status changed to ${data?.status}`,
+          );
+          abortController.abort();
+          pendingTxAbortControllers.delete(txId);
+        }
         continue;
+      }
+
+      if (data.type === TxType.CCTP_TO_AGORIC) continue;
+
       mustMatch(data, PublishedTxShape, `${path} index -1`);
       const tx = { txId, ...data } as PendingTx;
       log('New pending tx', tx);
+
+      const abortController = new AbortController();
+      pendingTxAbortControllers.set(txId, abortController);
```

Comment:
Given that we would never want to overwrite an existing entry in `pendingTxAbortControllers`, both `.set` calls should be replaced. My recommendation:
```suggestion
      provideLazyMap(
        pendingTxAbortControllers,
        txId,
        () => makeAbortController(),
      );
```